G1ANT.Addon.Net/API/SimplifiedMessageSummary.cs

```diff
@@ -122,11 +122,14 @@ public string MessageId
         {
             get
             {
-                if (fullMessage != null)
-                    return fullMessage.MessageId;
-                else if (messageSummary != null)
+                if (!string.IsNullOrEmpty(messageSummary?.Envelope?.MessageId))
```

Comment:
for optimisation purposes please keep previous flow
- first check fullMessage?.MessageId
- then fullMessage.Headers
- at the end messageSummary?.Envelope?.MessageId
- it is important to use private class variable rather than property

Reply:
I can change for sequence lines. 
I can't accept change from public property into private, because in that solution fullMessage always return null and in summary MessageId will alawys return empty string. I specially changed to property, because in this property is possibly creating instance for private variable with Folder.GetMessage(messageSummary.UniqueId)